packages/playback-core/src/index.ts

```diff
@@ -41,9 +49,143 @@ export {
+
+      return fetch(mediaPlaylistUrl)
+        .then((resp) => resp.text())
+        .then((mediaPlaylistStr) => mediaPlaylistStr.split('\n'));
+    });
+};
+
+export const getStreamInfoFromPlaylistLines = (playlistLines: string[]) => {
+  const typeLine = playlistLines.find((line) => line.startsWith('#EXT-X-PLAYLIST-TYPE')) ?? '';
+  const playlistType = typeLine.split(':')[1]?.trim() as HlsPlaylistTypes;
+  const streamType = toStreamTypeFromPlaylistType(playlistType);
+  const targetLiveWindow = toTargetLiveWindowFromPlaylistType(playlistType);
+
+  // Computation of the live edge start offset per media-ui-extensions proposal. See: https://github.com/video-dev/media-ui-extensions/blob/main/proposals/0007-live-edge.md#recommended-computation-for-rfc8216bis12-aka-hls (CJP)
+  let liveEdgeStartOffset = undefined;
+
+  if (streamType === StreamTypes.LIVE) {
+    // Required if playlist contains one or more EXT-X-PART tags. See: https://datatracker.ietf.org/doc/html/draft-pantos-hls-rfc8216bis-12#section-4.4.3.7 (CJP)
+    const partInfLine = playlistLines.find((line) => line.startsWith('#EXT-X-PART-INF'));
+    const lowLatency = !!partInfLine;
+
+    if (lowLatency) {
+      // The EXT-X-PART-INF only has one in-spec named attribute, PART-TARGET, which is required,
+      // so parsing & casting presumptuously here. See spec link above for more info. (CJP)
+      const partTarget = +partInfLine.split(':')[1].split('=')[1];
+      liveEdgeStartOffset = partTarget * 2;
+    } else {
+      // This is required for all media playlists. See: https://datatracker.ietf.org/doc/html/draft-pantos-hls-rfc8216bis-12#section-4.4.3.1 (CJP)
+      const targetDurationLine = playlistLines.find((line) => line.startsWith('#EXT-X-TARGETDURATION')) as string;
+      // EXT-X-TARGETDURATION has exactly one unnamed attribute that represents the target duration value, which is required,
+      // so parsing and casting presumptuously here. See spec link above for more info. (CJP)
+      const targetDuration = +targetDurationLine.split(':')[1];
+      liveEdgeStartOffset = targetDuration * 3;
+    }
+  }
+
+  return {
+    streamType,
+    targetLiveWindow,
+    liveEdgeStartOffset,
+  };
+};
+
+export const updateStreamInfoFromSrc = async (src: string, mediaEl: HTMLMediaElement) => {
+  const playlistLines = await getMediaPlaylistLinesFromMultivariantPlaylistSrc(src);
+
+  const { streamType, targetLiveWindow, liveEdgeStartOffset } = getStreamInfoFromPlaylistLines(playlistLines);
+
+  (muxMediaState.get(mediaEl) ?? {}).liveEdgeStartOffset = liveEdgeStartOffset;
+
+  (muxMediaState.get(mediaEl) ?? {}).targetLiveWindow = targetLiveWindow;
+  mediaEl.dispatchEvent(
+    new CustomEvent('targetlivewindowchange', { composed: true, bubbles: true, detail: targetLiveWindow })
+  );
+
+  (muxMediaState.get(mediaEl) ?? {}).streamType = streamType;
+  mediaEl.dispatchEvent(new CustomEvent('streamtypechange', { composed: true, bubbles: true, detail: streamType }));
```

Comment:
I wouldn't add `detail` here because that will never be added to the native HTML video tag